staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -52,6 +52,16 @@ import (
 // object.
 type ObjectFunc func(obj runtime.Object) error
 
+// ValidateObjectFunc is a function to act on a given object. An error may be returned
+// if the hook cannot be completed. An ObjectFunc may NOT transform the provided
+// object.
+type ValidateObjectFunc func(obj runtime.Object) error
+
+// ValidateObjectUpdateFunc is a function to act on a given object and its predecessor.
+// An error may be returned if the hook cannot be completed. An UpdateObjectFunc
+// may NOT transform the provided object.
+type ValidateObjectUpdateFunc func(obj, old runtime.Object) error
```

Comment:
counter-intuitive order. but it looks like we have a mix already today.

Reply:
> counter-intuitive order. but it looks like we have a mix already today.

It's the order from validators and strategies.